### PR TITLE
Remove partitionKey from RequestOptions

### PR DESCRIPTION
**Pull request by southpolesteve**
#320

### CODE REVIEW

File: src/client/Item/Item.ts [hunk 1]
```diff
@@ -28,7 +29,7 @@ export class Item {
   constructor(
     public readonly container: Container,
     public readonly id: string,
-    public readonly partitionKey: string,
+    public partitionKey: PartitionKey,
```
Comment (christopheranderson) [inline src/client/Item/Item.ts]: We shouldn't make this public settable. We should use a _partitionKey or partitionKeyInternal with a getter instead.

Comment (southpolesteve) [reply src/client/Item/Item.ts]: I made it private and it only gets public set in the constructor

**Comment by BernhardHofmann** [comment]
The [documentation for create](https://azure.github.io/azure-cosmos-js/classes/items.html#create) still states that the `RequestOptions` are the way to specify the partition key on a create. I can't find from issue #320 or this PR where or how I specify the partition key when creating an item now. 😞

I've submitted an issue for the documentation here: https://github.com/MicrosoftDocs/azure-docs/issues/83600

**Pull request merged**